core/src/main/java/org/n52/iceland/ogc/filter/ComparisonFilter.java

```diff
@@ -0,0 +1,315 @@
+     * @param valueUpper
+     *            upper value
+     * @throws OwsExceptionReport
+     *             If operator is not
+     *             {@link ComparisonOperator#PropertyIsBetween}
+     */
+    public ComparisonFilter(ComparisonOperator operator, String valueReference, String value, String valueUpper)
+            throws OwsExceptionReport {
+        super(valueReference);
+        if (operator == ComparisonOperator.PropertyIsBetween) {
+            this.operator = operator;
+            this.value = value;
+            this.valueUpper = valueUpper;
+        } else {
+            throw new NoApplicableCodeException()
+                    .withMessage("Use other constructor for ComparisonFilter! This constructor could only"
+                            + "be used for operator 'PropertyIsBetween'");
+        }
+    }
+
+    /**
+     * constructor for {@link ComparisonOperator#PropertyIsLike} filter
+     *
+     * @param operator
+     *            Filter operator
+     * @param valueReference
+     *            valueReference
+     * @param value
+     *            value
+     * @param valueUpper
+     *            upper value for between filter
+     * @param escapeString
+     *            Escape characters
+     * @throws OwsExceptionReport
+     *             If operator is not {@link ComparisonOperator#PropertyIsLike}
+     */
+    public ComparisonFilter(ComparisonOperator operator, String valueReference, String value, String valueUpper,
+            String escapeString) throws OwsExceptionReport {
+        super(valueReference);
+        if (operator == ComparisonOperator.PropertyIsLike) {
+            this.operator = operator;
+            this.value = value;
+            this.valueUpper = valueUpper;
+            this.escapeString = escapeString;
+        } else {
+            throw new NoApplicableCodeException()
+                    .withMessage("Use other constructor for ComparisonFilter! This constructor could only be used for operator 'PropertyIsLike'");
+        }
+    }
+
+    @Override
+    public ComparisonOperator getOperator() {
+        return operator;
+    }
+
+    @Override
+    public ComparisonFilter setOperator(ComparisonOperator operator) {
+        this.operator = operator;
+        return this;
+    }
+
+    /**
+     * Get filter value
+     *
+     * @return filter value
+     */
+    public String getValue() {
+        return value;
+    }
+
+    /**
+     * Get upper filter value
+     *
+     * @return upper filter value
+     */
+    public String getValueUpper() {
+        return valueUpper;
+    }
+
+    /**
+     * Set filter value
+     *
+     * @param value
+     *            filter value
+     */
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    /**
+     * Set upper filter value
+     *
+     * @param valueUpper
+     *            upper filter value
+     */
+    public void setValueUpper(String valueUpper) {
+        this.valueUpper = valueUpper;
```

Comment:
But all your constructor validation is worthless as the object is mutable. I can call `setValueUpper()` for all operators ;)